XenAdmin/MainWindow.cs

```diff
@@ -2666,16 +2666,27 @@ private void Updates_CollectionChanged(CollectionChangeEventArgs e)
 
         private void UpdatesCheck_Completed(bool suceeded, string err)
         {
-            
-            Program.Invoke(this, () => {
+
+            Program.Invoke(this, () =>
+            {
                 updateAlert = Updates.UpdateAlerts.FirstOrDefault(update => update is ClientUpdateAlert) as ClientUpdateAlert;
-                updateClientToolStripMenuItem.Enabled = updateAlert != null; 
+                DisableAndHideUpdateClientToolStripMenuItem(updateAlert != null);
             });            
         }
 
+        /// <summary>
+        /// Hides and disables the update client button

```

Comment:
super-nit: technically not what it does, more like "Changes state and visibility of the update client button"